### PR TITLE
feat(vue): support generating components using the path as provided

### PR DESCRIPTION
**Pull request by jaysoo**
This PR updates `@nx/vue:component` to support a component path provided as is.

e.g.

```
# this generates at libs/ui/src/button.vue, the path belongs to a project
nx g component libs/ui/src/button
```

### CODE REVIEW

File: packages/vue/src/generators/component/schema.json [hunk 1]
```diff
@@ -22,9 +22,7 @@
       "alias": "p",
       "$default": {
         "$source": "projectName"
-      },
-      "x-prompt": "What is the name of the project for this component?",
-      "x-priority": "important"
+      }
```
Comment (leosvelperez) [inline packages/vue/src/generators/component/schema.json:25]: We need to deprecate `--project`.

File: packages/vue/src/generators/component/schema.d.ts
Comment (leosvelperez) [inline packages/vue/src/generators/component/schema.d.ts:1]: We need to add the deprecated JSDoc for the deprecated properties here to notify devs using the generator programmatically.

File: packages/vue/src/generators/component/schema.json
Comment (leosvelperez) [inline packages/vue/src/generators/component/schema.json:1]: We also need to deprecate `--pascalCaseFiles` and `--pascalCaseDirectory`.

Comment (leosvelperez) [inline packages/vue/src/generators/component/schema.json:1]: Let's also update the description for `--directory` to match the one we're adding for other generators using both formats.

**Review by leosvelperez — DISMISSED** [review]

**Review by mandarini — APPROVED** [review]

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**